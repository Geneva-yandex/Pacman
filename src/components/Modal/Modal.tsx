import * as React from 'react';
import bem from 'easy-bem';

const b = bem('Modal');

type Props = {
    isClosed: boolean,
    className?: string,
    close: (event: React.MouseEvent<HTMLElement>) => void,
    stateModal: string,
};

export default class Modal extends React.Component<Props> {
    render() {

        const {className, isClosed, close, stateModal} = this.props;

        if (isClosed) {
            return null;
        }

        return (
            <div className={'modal'}>
                <div className={className}>
                    <section className={b('wrapper')}>
                        <div onClick={close} className="close-button" data-modal={stateModal}>X
                        </div>
                        {this.props.children}
                    </section>
                </div>
            </div>
        );
    }
}
