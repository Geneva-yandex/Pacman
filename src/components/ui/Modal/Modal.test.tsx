import * as React from 'react';
import renderer from 'react-test-renderer';
import Modal from './Modal';

const props = {
    isClosed: false,
    close: () => {},
    stateModal: 'test'
};
describe('Modal', () => {
    test('Modal component renders correctly', () => {
        const renderedComponent = renderer
            .create(<Modal {...props}/>)
            .toJSON();

        expect(renderedComponent).toMatchSnapshot();
    });
});
