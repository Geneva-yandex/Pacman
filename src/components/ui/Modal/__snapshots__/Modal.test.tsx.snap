// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Modal component renders correctly 1`] = `
<div
  className="modal"
>
  <div>
    <section
      className="Modal__wrapper"
    >
      <div
        className="close-button"
        data-modal="test"
        onClick={[Function]}
      >
        X
      </div>
    </section>
  </div>
</div>
`;
